src/v0/destinations/braze/transform.js

```diff
@@ -148,38 +225,81 @@ function getUserAttributesObject(message, mappingJson) {
         }
       }
     });
+
+    // populate data with custom attribute operations
+    populateCustomAttributesWithOperation(
+      traits,
+      data,
+      message.properties?.mergeObjectsUpdateOperation,
+      destination?.Config.enableNestedArrayOperations,
+    );
   }
 
   return data;
 }
 
-function processIdentify(message, destination) {
+async function processIdentify(message, destination) {
   // override userId with externalId in context(if present) and event is mapped to destination
   const mappedToDestination = get(message, MappedToDestinationKey);
   if (mappedToDestination) {
     adduserIdFromExternalId(message);
   }
 
-  return buildResponse(
-    message,
-    destination,
-    getIdentifyPayload(message),
-    getIdentifyEndpoint(destination.Config.endPoint),
-  );
+  const identifyPayload = getIdentifyPayload(message);
+  const identifyEndpoint = getIdentifyEndpoint(getEndpointFromConfig(destination));
+
+  const brazeIdentifyResp = await httpPOST(identifyEndpoint, identifyPayload, {
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+      Authorization: `Bearer ${destination.Config.restApiKey}`,
+    },
+  });
+
+  const processedBrazeIdentifyResp = processAxiosResponse(brazeIdentifyResp);
+  if (!isHttpStatusSuccess(processedBrazeIdentifyResp.status)) {
+    throw new NetworkError(
+      'Braze identify failed',
+      processedBrazeIdentifyResp.status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(processedBrazeIdentifyResp.status),
+      },
+      processedBrazeIdentifyResp.response,
+    );
+  }
 }
 
-function processTrackWithUserAttributes(message, destination, mappingJson) {
+function processTrackWithUserAttributes(message, destination, mappingJson, deduplicationStore) {
   let payload = getUserAttributesObject(message, mappingJson);
   if (payload && Object.keys(payload).length > 0) {
     payload = setExternalIdOrAliasObject(payload, message);
+    const requestJson = { attributes: [payload] };
+    if (destination.Config.deduplicationEnabled) {
+      const dedupedAttributePayload = BrazeDedupUtility.deduplicate(
+        payload,
+        deduplicationStore.userStore,
+      );
+      if (
+        isDefinedAndNotNullAndNotEmpty(dedupedAttributePayload) &&
+        Object.keys(dedupedAttributePayload).some(
+          (key) => !['external_id', 'user_alias'].includes(key),
+        )
+      ) {
+        requestJson.attributes = [dedupedAttributePayload];
+      } else {
+        delete requestJson.attributes;
```

Comment:
in this case, do we need to make braze delivery call?

Reply:
fixed